src/system/user.cr

```diff
@@ -0,0 +1,136 @@
+  end
+
+  # Returns the user name for the given user ID.
+  # Returns `nil` if no user exists.
+  def self.name?(uid : Int) : String?
+    check_uid_in_bounds(uid)
+    user_struct = LibC.getpwuid(uid)
+    return nil if !user_struct
+    String.new(user_struct.value.pw_name)
+  end
+
+  # Returns the user ID for the given user name.
+  # Raises `System::User::NotFoundError` if no user exists.
+  def self.uid(user : String) : UInt32
+    uid?(user) || raise NotFoundError.new("User #{user.inspect} was not found")
+  end
+
+  # Returns the user ID for the given user name.
+  # Returns `nil` if no user exists.
+  def self.uid?(user : String) : UInt32?
+    user.check_no_null_byte
+    user_struct = LibC.getpwnam(user)
+    return nil if !user_struct
+    user_struct.value.pw_uid
+  end
+
+  # Returns the user specified by the user ID.
+  # Raises `System::User::NotFoundError` if not found.
+  def self.get(uid : Int) : User
+    get?(uid) || raise NotFoundError.new("User #{uid.inspect} was not found")
+  end
+
+  # Returns the user specified by the username.
+  # Returns `nil` if not found.
+  def self.get?(uid : Int) : User?
+    check_uid_in_bounds(uid)
+    user_struct = LibC.getpwuid(uid)
+    return nil if !user_struct
+    new(user_struct.value)
+  end
+
+  # Returns the user specified by the username.
+  # Raises `System::User::NotFoundError` if not found.
+  def self.get(username : String) : User
+    get?(username) || raise NotFoundError.new("User #{username.inspect} was not found")
+  end
+
+  # Returns the user specified by the username.
+  # Returns `nil` if not found.
+  def self.get?(username : String) : User?
+    username.check_no_null_byte
+    user_struct = LibC.getpwnam(username)
+    return nil if !user_struct
+    new(user_struct.value)
+  end
+
+  # :nodoc:
+  private def initialize(user : LibC::Passwd)
+    @name = String.new(user.pw_name)
+    @uid = user.pw_uid
+    @gid = user.pw_gid
+    @home = String.new(user.pw_dir)
+    @shell = String.new(user.pw_shell)
+    @info = String.new(user.pw_gecos)
+  end
+
+  # Returns the users name as a `String`.
+  getter name : String
+
+  # Returns the users ID.
+  getter uid : UInt32
+
+  # Returns the users group ID.
+  getter gid : UInt32
+
+  # Returns the path for the users home as a `String`.
+  getter home : String
+
+  # Returns the path for the users shell as a `String`.
+  getter shell : String
+
+  # Returns additional information about the user as a `String`.
+  getter info : String
+
+  # Returns the primary `Group` for the user.
+  def group
+    Group.get(@gid)
+  end
+
+  # See `Object#hash(hasher)`
+  def hash(hasher)
+    @uid.hash(hasher)
+  end
+
+  # Compares this user with *other*, returning `-1`, `0` or `+1` depending if the
+  # user ID is less, equal or greater than the *other* user ID.
+  def <=>(other : User) : Int
+    @uid <=> other.uid
+  end
+
+  # Returns a `String` representation of the user. Its user name.
```

Comment:
Again, the last sentence. I don't think this is a good solution, misses a verb. Could work when separates by a comma. Otherwise the same suggestion as with the group method. At least, both should be similar.

Reply:
Fixed